### DISCUSSION TITLE
Units degC

### DISCUSSION THREAD

**Discussion by tomelgato** [davetcc/tcMenu#464] [General]
Is there a way to do °C as unit in AnalogMenuItem?

**Comment by davetcc** [comment]
Could you try using the character code for degrees directly into the file. Honestly, I've never tried a non-ascii UTF-8 character there, but I've seen images of menus with a Kanji character there, so must be possible.

**Comment by tomelgato** [comment]
Didnt find a solution till now to do it with TCMenu Gui

		lcd.println("\xDF");

does show the deg character but its not possible to enter a \ in tcmenu.

When i try to use the ascii character by mac it would be 

		lcd.println("ﬂ");

but this character just outputs garbage, either direct or via tcmenu

P.S. you can copy a hex char value to the clipboard with osx terminal:
printf "\xDF" | pbcopy

**Comment by tomelgato** [comment]
The library can handle the chars, with a call like that it works and shows °C:

const PROGMEM AnalogMenuInfo minfoTemp = { "Temp", 28, 0xffff, 150, Temp, 0, 1, "\xDF\x43" };

even better in octal notation, then its easy to continue with normal characters:

const PROGMEM AnalogMenuInfo minfoTemp = { "Temp", 28, 0xffff, 150, Temp, 0, 1, "\337C" };


So without thinking about other consequences it would be sufficient if TCMenu allows the \ char

**Reply by davetcc** [reply]
Agreed, C++ default is to use UTF-8, and Java's default output to a text file is also UTF-8. The only issue is that the UI does not handle this very well. I'll try and take a look at quick solutions.

**Reply by davetcc** [reply]
See #475, will be in  4.3

**Reply by davetcc** [reply]
Closing this out, supported in desktop, web designer should handle this out of the box as using UTF-8
